server/routes/getting_started/getting_started_router.ts

```diff
@@ -134,4 +129,67 @@ export function registerGettingStartedRoutes(router: IRouter) {
       }
     }
   );
+
+  router.post(
+    {
+      path: `/api/observability/gettingStarted/mdsLoading`,
+      validate: {
+        body: schema.object({
+          mdsId: schema.string(),
+          mdsLabel: schema.string(),
+          tutorialId: schema.string(),
+        }),
+      },
+    },
+    async (
+      context,
+      request,
+      response
+    ): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+      try {
+        const { mdsId, mdsLabel, tutorialId } = request.body;
+        const fileData = await loadAssetsFromFile(tutorialId);
+
+        const objects = await createSavedObjectsStreamFromNdJson(Readable.from(fileData));
+        const loadedObjects = await objects.toArray();
+
+        const updatedObjects = loadedObjects.map((obj) => {
+          if (mdsId) {
+            const newId = `${mdsId}${obj.id}`;
+            const newReferences =
+              obj.references && obj.references.length > 0
+                ? obj.references
```

Comment:
1. We still need to add datasource here, even if the object contains reference. Are we doing it here?
2. Dashboards contains references to visualizations, in that case are you also changing the id's to visualizations with MDSId prefrix and updating their references?

Reply:
Updated code to now always add the datasource reference to all objects with a reference. Also updates the id of the visualizations to match the change.
//Visualization 
<img width="840" alt="Screenshot 2024-08-09 at 12 00 32 PM" src="https://github.com/user-attachments/assets/9ad35367-9a30-45b0-a179-05ed3e950d79">
//Non visualization
<img width="803" alt="Screenshot 2024-08-09 at 12 00 57 PM" src="https://github.com/user-attachments/assets/4d772973-062f-42b2-8341-74eddcf07517">